lib/parser/provenance.cc

```diff
@@ -69,6 +69,18 @@ ProvenanceRange OffsetToProvenanceMappings::Map(std::size_t at) const {
   return provenanceMap_[low].range.Suffix(offset);
 }
 
+std::optional<std::size_t> OffsetToProvenanceMappings::ReverseMap(
+    Provenance at) const {
+  for (unsigned int i = 0; i < provenanceMap_.size(); i++) {
+    if (provenanceMap_[i].range.Contains(at)) {
+      std::size_t offset{
+          at.offset() - provenanceMap_[i].range.start().offset()};
+      return {provenanceMap_[i].start + offset};
```

Comment:
The braces are probably not necessary.

Reply:
I see some examples where braces are used in this way, such as CookedSource::GetProvenanceRange. Can you explain to me when it should/shouldn't be used?

Reply:
They're not needed for `std::optional<>` results.  Some old versions of C++ compilers &/or STL header files cause bogus errors to be emitted for constructors of C++17 template classes and the braces can be used as workarounds.

Reply:
Thank you.